src/main/java/org/primefaces/component/datatable/export/DataTableExporter.java

```diff
@@ -379,4 +384,24 @@ public int invoke(FacesContext context) {
             return counter;
         }
     }
+
+    protected void setResponseHeader(ExternalContext externalContext , String contentDisposition) {
+        externalContext.setResponseHeader("Expires", "0");
+        externalContext.setResponseHeader("Cache-Control", "must-revalidate, post-check=0, pre-check=0");
+        externalContext.setResponseHeader("Pragma", "public");
+        externalContext.setResponseHeader("Content-disposition", contentDisposition);
+    }
+
+    protected void addResponseCookie(ExternalContext externalContext) {
+        final boolean secure = ((HttpServletRequest) externalContext.getRequest()).isSecure();
+        if (secure) {
+            Map<String, Object> map = new HashMap<String, Object>();
+            map.put("secure", secure);
+            externalContext.addResponseCookie(Constants.DOWNLOAD_COOKIE, "true", map);
+        }
+        else {
+            externalContext.addResponseCookie(Constants.DOWNLOAD_COOKIE, "true", Collections.<String, Object>emptyMap());
+        }
+    }
+
```

Comment:
I think a better way to write this method is:
```java
        protected void addResponseCookie(ExternalContext externalContext) {
            final boolean secure = ((HttpServletRequest) externalContext.getRequest()).isSecure();
            Map<String, Object> map;
            if (secure) {
                map = new HashMap<>(1);
                map.put("secure", secure);
            }
            else {
                map = Collections.emptyMap();
            }
            externalContext.addResponseCookie(Constants.DOWNLOAD_COOKIE, "true", map);
        }
```

Reply:
> 1. you must also take care of PortletRequest. We already have this check somewhere and reuseable.

Had to make a minor refactor to support passing an ExternalContext

> 2. AFAIR the Cookie is also created/removed on clientside